main.tf

```diff
@@ -104,13 +104,13 @@ module "container_definition" {
 
 locals {
   alb = {
-    container_name   = coalesce(var.alb_container_name, module.this.id)
+    container_name   = module.this.enabled ? coalesce(var.alb_container_name, module.this.id) : null
     container_port   = var.container_port
```

Comment:
Why do the container names also need to use the enable logic

Reply:
AFAIK `coalesce` has to return a not-null value and if this module is disabled and `var.alb_container_name` is not set (it's optional), this line will raise an error.

I could replace it with `try` if you don't like this approach:
```tf
container_name = try(coalesce(var.alb_container_name, module.this.id), null)
```

Reply:
Can we update the line to drop coalesce then?

```
    container_name   = var.alb_container_name != null ? var.alb_container_name : module.this.id
```

Reply:
Done 👍🏻